discord/voice_client.py

```diff
@@ -187,7 +187,7 @@ async def _create_socket(self, server_id, data):
                         'If timeout occurred considering raising the timeout and reconnecting.')
             return
 
-        self.endpoint = endpoint.replace(':80', '')
+        self.endpoint = endpoint.replace(':80', '').replace(':443', '')
```

Comment:
Surely `endpoint.split(':')[0]` is more pythonic?

Reply:
.split allocates a list. `.rpartition(':')[0]` would be more efficient (and support ipv6 hostnames with :80 in their addresses)

Reply:
But split would remove the error forever doesn't it? and what happens if this error occurs again but with 81 as an example?

Reply:
This has already been fixed with https://github.com/Rapptz/discord.py/commit/a9824ffabf0b2c8031ac75245650711145aaa26c in v1.4.0 and newer.